### PR TITLE
Prepare for a 0.13.0 release

### PR DESCRIPTION
**Pull request by joshuagl**
**Fixes issue #**: N/A

**Description of the changes being introduced by the pull request**:
* Add a document describing the release process
* prepare docs/CHANGELOG.md, setup.cfg and tuf/__init__.py for a 0.13.0 release

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

File: docs/RELEASE.md
Comment (joshuagl) [inline docs/RELEASE.md:11]: Can't currently perform this step, as @vladimir-v-diaz is the owner of that package on test.pypi.org

File: docs/RELEASE.md
Comment (lukpueh) [inline docs/RELEASE.md]: Very minor nit: Could you escape the double underscore in `__init__.py` (or use backticks)?

File: docs/RELEASE.md
Comment (lukpueh) [inline docs/RELEASE.md]: Very very minor (consistency) nit:
```suggestion
  * Remove existing dist build dirs
  * Create source dist `python setup.py sdist`
  * Create wheel (with 2 and 3 support) `python setup.py bdist_wheel --universal`
```

File: docs/RELEASE.md
Comment (lukpueh) [inline docs/RELEASE.md]: Very very minor (consistency) nit:
```suggestion
  * Remove existing dist build dirs
  * Create source dist `python setup.py sdist`
  * Create wheel (with 2 and 3 support) `python setup.py bdist_wheel --universal`
```

File: docs/RELEASE.md
Comment (lukpueh) [inline docs/RELEASE.md]: Could we make this two lines?  E.g.
```suggestion
  * Sign source dist `gpg --detach-sign -a dist/tuf-vA.B.C.tar.gz`
  * Sign wheel `gpg --detach-sign -a dist/tuf-vA.B.C-py2.py3-none-any.whl`
```

File: docs/RELEASE.md
Comment (lukpueh) [inline docs/RELEASE.md]:
```suggestion
  `git tag --sign vA.B.C`
```

**Review by lukpueh — CHANGES_REQUESTED** [review]
Many thanks for preparing this release, and big kudos for adding a `RELEASE.md` document, @joshuagl. ❤️ 
The former definitely looks good to me. I double-checked that all PR numbers point to the right PR and no important PRs were omitted. You only left out dependabot PRs and a few minor changes that probably could be added to the changelog but don't need to, i.e. #1021 #1028 #1044 #1051 #1072 #1086. Maybe double-check if you left them out on purpose or not.

In RELEASE.md I left a few comments, feel free to address as you see fit. Thanks!

**Comment by trishankatdatadog** [comment]
> Many thanks for preparing this release, and big kudos for adding a `RELEASE.md` document, @joshuagl. ❤️

💯 💯 

QQ: would it be possible to formalize `RELEASE.md` with a script?

**Comment by trishankatdatadog** [comment]
I would also probably hold off until we merge some important fixes, including #1091

File: docs/RELEASE.md
Comment (joshuagl) [reply docs/RELEASE.md]: Whoops, thanks for catching that!

**Comment by joshuagl** [comment]
> Many thanks for preparing this release, and big kudos for adding a `RELEASE.md` document, @joshuagl. ❤️
> The former definitely looks good to me. I double-checked that all PR numbers point to the right PR and no important PRs were omitted. You only left out dependabot PRs and a few minor changes that probably could be added to the changelog but don't need to, i.e. #1021 #1028 #1044 #1051 #1072 #1086. Maybe double-check if you left them out on purpose or not.
> 
> In RELEASE.md I left a few comments, feel free to address as you see fit. Thanks!

I may have missed, #1072, but mostly those PRs were things I opted to ignore because I didn't think they were particularly interesting. I'd be happy to include the PRs you mentioned, or not, and add some additional text to `RELEASE.md` to guide future `CHANGELOG.md` editors in curating a set of changes that better matches expectations. 

Personally I prefer a changelog to include __notable__ changes, rather than all changes. However I realise even that is subjective and there are entries I added to the changelog for this release that don't meet my own criteria for notable.

I welcome thoughts on changelog curation. In the past I've followed similar principles to https://keepachangelog.com/, which would suggest editing the changelog here to group similar changes and clearly delineate new features, changed features, fixes, security issues, etc.

**Comment by joshuagl** [comment]
> > Many thanks for preparing this release, and big kudos for adding a `RELEASE.md` document, @joshuagl. ❤️
> 
> 💯 💯
> 
> QQ: would it be possible to formalize `RELEASE.md` with a script?

Agreed, scripting some of this would be good. I'll take a look at that in future.

**Comment by joshuagl** [comment]
> I would also probably hold off until we merge some important fixes, including #1091

I'm hoping to get into the habit of releasing more regularly. docs/ROADMAP.md suggests the project releases every 3 months. I'd like to get closer to that, so I don't think in general we should worry too much about holding a release for fixes.

#1091 specifically, though, I'd be happy to merge and include in the release

**Comment by joshuagl** [comment]
> #1091 specifically, though, I'd be happy to merge and include in the release

I've merged this PR and included it in `docs/CHANGELOG.md`

**Comment by lukpueh** [comment]
> I may have missed, #1072, but mostly those PRs were things I opted to ignore because I didn't think they were particularly interesting. I'd be happy to include the PRs you mentioned, or not, and add some additional text to `RELEASE.md` to guide future `CHANGELOG.md` editors in curating a set of changes that better matches expectations.
> 
> Personally I prefer a changelog to include **notable** changes, rather than all changes. However I realise even that is subjective and there are entries I added to the changelog for this release that don't meet my own criteria for notable.
> 
> I welcome thoughts on changelog curation. In the past I've followed similar principles to https://keepachangelog.com/, which would suggest editing the changelog here to group similar changes and clearly delineate new features, changed features, fixes, security issues, etc.

👍  This all makes sense and aligns with what I thought. Just wanted to double-check. :) Thanks for adding a link to the changelog guide to RELEASE.md.

**Review by lukpueh — APPROVED** [review]
💯

**Pull request merged**
